Scheduler: Changing drop target calculation when dragging appointment

vsimonaitis opened the pull request:
The issue derived from support ticket https://www.devexpress.com/Support/Center/Question/Details/T728235/when-dragging-events-in-timelinemonth-boxes-shifts

Problem
In current implementation, when dragging events in scheduler, dropping point is always under mouse cursor. With small draggable items it is sufficient to drop under cursor, but when event box is wide, drop area under mouse cursor becomes unintuitive.

To reproduce
See demo https://codepen.io/anon/pen/RdXqJv
When grabbing a multi-day event (3 days or more) in scheduler month view grab the bottom-right corner. 
Slightly move the mouse (mouse may stay in the same cell area) and release it. Notice that events start day is shifted by events initial duration. What user would expect is that accidental minor drag would not change the event and drop it to the same date cell. 

Fix
The fix changes calculation of dropping point for scheduler event boxes. Instead of dropping box under the mouse cursor, the item is dropped under its top-left edge position. This is way more intuitive for the users.

<!--
Make sure that you've read the "Contributing Code and Content" section in CONTRIBUTING.md

If you are submitting a bug fix, the subject should contain "fixes ISSUE_ID" or "resolves ISSUE_ID".

In addition, please clarify:

1. What did you do? 
2. How did you do it?
3. How should we verify this?

-->

Commit:
Changing drop target calculation when dragging

vsimonaitis committed:
Removing trailing spaces

andrewmakarov reviewed (commented):
Your request breaks existing tests and doesn't contain any new tests. So it's not clear how it works in other existing scenarios. We cannot accept it in this format.

Commit:
Reverting global drag-drop behaviour change and localizing alteration only to scheduler

Commit:
Ensuring we get jQuery object

Commit:
Dragging appointment down instead of up in test

vsimonaitis committed:
Adding new lines to completely revert to previous version

vsimonaitis committed:
Removing whitespaces

vsimonaitis commented on testing/tests/DevExpress.ui.widgets.scheduler/appointments.tests.js:
```diff
@@ -543,7 +543,7 @@ QUnit.test("Allday appointment should stay in allDayContainer after small draggi
         pointer = pointerMock($appointment).start();
 
-    pointer.dragStart().drag(0, -30);
+    pointer.dragStart().drag(0, 30);
```
Drop target is anchored under top-left corner. Dragging the appointment up would move the appointment to another row, so I had to change the dragging direction.

vsimonaitis commented on js/ui/scheduler/ui.scheduler.appointments.js:
```diff
@@ -818,6 +823,7 @@ var SchedulerAppointments = CollectionWidget.inherit({
             },
             onDrag: function(args) {
                 correctCoordinates(args.element);
+                adjustDropPosition(args.event, args.element);
```
Adjusting drop target anchor by the position of top-left corner of appointment being dragged.

vsimonaitis commented:
I have changed the code and fixed the UI tests. The drag and drop behavior change is now localized to Scheduler only and is not affecting other generic drag and drop functionality.

vsimonaitis commented:
@andrewmakarov Please review my last changes

andrewmakarov commented:
> @andrewmakarov Please review my last changes

Hi,

I reviewed your code and found that the suggested fix won't work in certain scenarios. We're going to address this issue in  a new pull request that should cover all problems. I'll make an update here once it's ready.

andrewmakarov commented:
> @andrewmakarov Please review my last changes

We're going to address this issue #9617 :
I'm closing this request
